session/txnmanager.go

```diff
@@ -134,6 +135,14 @@ func (m *txnManager) OnStmtErrorForNextAction(point sessiontxn.StmtErrorHandlePo
 	return m.ctxProvider.OnStmtErrorForNextAction(point, err)
 }
 
+// ActivateTxn decides to activate txn according to the parameter `active`
+func (m *txnManager) ActivateTxn() (kv.Transaction, error) {
+	if m.ctxProvider == nil {
+		return nil, errors.AddStack(kv.ErrInvalidTxn)
+	}
+	return m.ctxProvider.ActivateTxn(nil)
```

Comment:
Passing nil value is confusing, can we save the type in baseProvider?

Reply:
Yes, it should be better.